kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/KubernetesKitConfiguration.java

```diff
@@ -35,16 +36,27 @@
 import com.vaadin.kubernetes.starter.sessiontracker.serialization.SerializationDebugRequestHandler;
 import com.vaadin.kubernetes.starter.sessiontracker.serialization.SpringTransientHandler;
 import com.vaadin.kubernetes.starter.sessiontracker.serialization.TransientHandler;
+import com.vaadin.pro.licensechecker.LicenseChecker;
 
 /**
  * This configuration bean is provided to auto-configure Vaadin apps to run in a
  * clustered environment.
  */
+@ConditionalOnProperty(name = "auto-configure", prefix = KubernetesKitProperties.PREFIX, matchIfMissing = true)
 @AutoConfiguration(afterName = "org.springframework.boot.autoconfigure.data.redis.RedisAutoConfiguration")
 @EnableConfigurationProperties({ KubernetesKitProperties.class,
         SerializationProperties.class })
 public class KubernetesKitConfiguration {
 
+    static final String PRODUCT_NAME = "vaadin-kubernetes-kit";
+
+    static final String PRODUCT_VERSION = "1.0";
+
+    static {
+        LicenseChecker.checkLicenseFromStaticBlock(PRODUCT_NAME,
```

Comment:
Removing the static block the test doesn't fail. This could be tested if the check license is called, example with MockedStatic.

Reply:
Great idea!